src/errors/location.rs

```diff
@@ -182,6 +188,22 @@ impl Location {
             }
         };
     }
+
+    pub fn with_prefix(&self, prefix: &Location) -> Location {
```

Comment:
I'm a bit confused on how this is used, and why `prefix` is reversed here but `self` is not?